WPThemeReview/Sniffs/Templates/ReservedTemplateNameSniff.php

```diff
@@ -0,0 +1,156 @@
+
+	/**
+	 * Found prefix in a file
+	 *
+	 * @var string
+	 */
+	protected $prefix;
+
+	/**
+	 * File slug
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * List of reserved template file names.
+	 *
+	 * @link https://developer.wordpress.org/themes/basics/template-hierarchy/
+	 * @link https://developer.wordpress.org/themes/template-files-section/partial-and-miscellaneous-template-files/#content-slug-php
+	 * @link https://wphierarchy.com/
+	 * @link https://en.wikipedia.org/wiki/Media_type#Naming
+	 *
+	 * @since 0.2.0
+	 *
+	 * @var array
+	 */
+	protected $reserved_file_name_prefixes = [
+		'author-'   => true,
+		'category-' => true,
+		'page-'     => true,
+		'tag-'      => true,
+	];
+
+	/**
+	 * Strip quotes surrounding an arbitrary string.
+	 *
+	 * Intended for use with the contents of a T_CONSTANT_ENCAPSED_STRING / T_DOUBLE_QUOTED_STRING.
+	 *
+	 * Copied from the WordPressCS\WordPress\Sniff abstract class.
+	 *
+	 * @since 0.2.0
+	 *
+	 * @param string $string The raw string.
+	 * @return string String without quotes around it.
+	 */
+	private function strip_quotes( $string ) {
+		return preg_replace( '`^([\'"])(.*)\1$`Ds', '$2', $string );
+	}
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			\T_OPEN_TAG,
+			\T_OPEN_TAG_WITH_ECHO,
+		);
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param \PHP_CodeSniffer\Files\File $phpcsFile The PHP_CodeSniffer file where the
+	 *                                               token was found.
+	 * @param int                         $stackPtr  The position of the current token
+	 *                                               in the stack.
+	 *
+	 * @return void|int Optionally returns an integer stack pointer or void to continue
+	 *                  normal file processing.
+	 */
+	public function process( File $phpcsFile, $stackPtr ) {
+
+		// Usage of `strip_quotes` is to ensure `stdin_path` passed by IDEs does not include quotes.
+		$file = $this->strip_quotes( $phpcsFile->getFileName() );
+
+		$fileName = basename( $file );
+
+		if ( \defined( '\PHP_CODESNIFFER_IN_TESTS' ) ) {
+			$fileName = str_replace( '.inc', '.php', $fileName );
+		}
+
+		// Check if the current file has a prefix in the reserved list.
+		if ( ! $this->is_reserved_template_name_used( $fileName ) ) {
+			return ( $phpcsFile->numTokens + 1 );
+		}
+
+		$phpcsFile->addError(
+			self::ERROR_MSG,
+			0,
+			'ReservedTemplatePrefixFound',
+			array( $this->prefix, $this->slug )
+		);
+	}
+
+	/**
+	 * Checks if the given file is located in the $reserved_file_name_prefixes array.
+	 *
+	 * @param  string $file File name to check.
+	 * @return boolean
+	 */
+	private function is_reserved_template_name_used( $file ) {
```

Comment:
In the previous iteration of this method, you only looked at the part before the `-`, as you now also want to access/remember the part of the file name after the `-`, there is an even simpler way of doing this: using `explode()` with a limit.

If - like suggested above -, you strip off the `.php` from the filename before passing it to this function, the net result will be already exactly what you need. See the below code snippet (untested, but should work).

```php
$parts = explode( '-', $file, 2 );
if ( empty( $parts[0] ) || empty( $parts[2] ) ) {
	// No dash or dash at start or end of filename, i.e. `-something.php`.
	return false;
}

// $parts[0] will now contain what you call `$partial` in the original code,
// while `$parts[1]` will contain the rest of the file name.
```

Reply:
I'll change the `$reserved_file_name_prefixes` to include prefixes without the dash. This will solve the [above comment](https://github.com/WPTRT/WPThemeReview/pull/213#discussion_r292825884) as well 🙂